src/edu/csus/ecs/pc2/ui/EditProblemPane.java

```diff
@@ -3703,6 +3744,10 @@ public void mousePressed(MouseEvent e) {
         return lblWhatsThisCLICSValidator;
     }
 
+    /*
+     * TODO: Future development: All these strings have to be moved out of the source code and into files or resources or something.  This stuff does
+     * not belong in source code. (JB)
```

Comment:
I agree that (perhaps) the strings should be moved into their own class (it would certainly make reading THIS class, and ones like it, easier).  However, I'm not sure I agree that they should be moved (for example) into files.  External files could be edited by someone, resulting in "customization" of the messages in ways we would not be happy with (adding derogatory comments, for example).   Maintaining the strings in source code (although perhaps in separate files would require someone to edit the source and then rebuild the system.  (True, they could fork the project and do that, since it's open source.  But they couldn't alter the text being displayed by OUR DISTRIBUTION...)